File: music-api/src/main/java/com/music/sale/web/user/request/CreateUserByPhoneRequest.java
package com.music.sale.web.user.request;

import com.music.sale.domain.user.enums.Gender;
import com.music.sale.domain.user.enums.UserRole;
import com.music.sale.domain.user.enums.UserType;
import java.time.LocalDate;

public record CreateUserByPhoneRequest(
        String phoneNumber,
        String password,
        String name,
        String nickname,
        UserRole role,
        LocalDate birthDate,
        Gender gender,
        String zipcode,
        String baseAddress,
        String detailAddress,
        UserType userType,
        Boolean marketingAgreed) {

    public CreateUserByPhoneRequest {
        if (role == null) {
            role = UserRole.USER;
        }
        if (userType == null) {
            userType = UserType.BUYER;
        }
        if (marketingAgreed == null) {
            marketingAgreed = Boolean.FALSE;
        }
    }
}

File: music-api/src/main/kotlin/com/music/sale/web/user/request/CreateUserByPhoneRequest.kt
package com.music.sale.web.user.request

import com.music.sale.domain.user.enum.Gender
import com.music.sale.domain.user.enum.UserRole
import com.music.sale.domain.user.enum.UserType
import java.time.LocalDate

data class CreateUserByPhoneRequest(
    val phoneNumber: String,
    val password: String,
    val name: String,
    val nickname: String,
    val role: UserRole = UserRole.USER,
    val birthDate: LocalDate? = null,
    val gender: Gender? = null,
    val zipcode: String? = null,
    val baseAddress: String? = null,
    val detailAddress: String? = null,
    val userType: UserType = UserType.BUYER,
    val marketingAgreed: Boolean = false,
)
